internal/installcfg/installcfg.go

```diff
@@ -391,11 +391,14 @@ func (i *installConfigBuilder) getInstallConfig(cluster *common.Cluster, addRhCa
 
 		bootstrapCidr := network.GetPrimaryMachineCidrForUserManagedNetwork(cluster, i.log)
 		if bootstrapCidr != "" {
-			i.log.Infof("None-Platform: Selected bootstrap machine network CIDR %s for cluster %s", bootstrapCidr, cluster.ID.String())
-			cfg.Networking.MachineNetwork = []MachineNetwork{{Cidr: bootstrapCidr}}
-			cluster.MachineNetworks = network.CreateMachineNetworksArray(bootstrapCidr)
+			i.log.Infof("None-Platform or SNO: Selected bootstrap machine network CIDR %s for cluster %s", bootstrapCidr, cluster.ID.String())
+			machineNetwork := []MachineNetwork{}
+			for _, net := range network.GetMachineNetworksFromBoostrapHost(cluster, i.log) {
+				machineNetwork = append(machineNetwork, MachineNetwork{Cidr: string(net.Cidr)})
+			}
+			cfg.Networking.MachineNetwork = machineNetwork
+			cluster.MachineNetworks = network.GetMachineNetworksFromBoostrapHost(cluster, i.log)
```

Comment:
I'm struggling to understand all that is happening in this block of code. I don't think all of it has to do with your specific changes.

* Why are we calling the same function twice?
* Why do we call it a MachineNetwork in the installcfg package when we have a MachineNetwork from models? I guess one is the API facing representation versus the internal one? either way it's weird.

Here is a suggestion that makes sense to me, tell me if/why it's crazy and wouldn't work (or why you don't like it).

```suggestion
			i.log.Infof("None-Platform or SNO: Selected bootstrap machine network CIDR %s for cluster %s", bootstrapCidr, cluster.ID.String())
			machineNetwork := []MachineNetwork{}
			cluster.MachineNetworks = network.GetMachineNetworksFromBoostrapHost(cluster, i.log)
			for _, net := range cluster.MachineNetworks {
				machineNetwork = append(machineNetwork, MachineNetwork{Cidr: string(net.Cidr)})
			}
			cfg.Networking.MachineNetwork = machineNetwork
```

One more question. Why do we need, here, a `[]MachineNetwork{}` and a `[]*models.MachineNetwork{}`?

Reply:
>Why are we calling the same function twice?

That was stupid and is fixed now

>Why do we call it a MachineNetwork in the installcfg package when we have a MachineNetwork from models? I guess one is the API facing representation versus the internal one? either way it's weird.

Your guess is correct. If you look at `models.MachineNetwork` you can see it stores CIDR as well as clusterID because this is how we store it in the DB. On the other hand the `MachineNetwork` from the `installcfg` package is a local struct used as a wrapper around a single property (string, CIDR) which is used in the generation of install-config.yaml.

`models.MachineNetwork` has been created very very recently whereas `MachineNetwork` from the `installcfg` exists since forever. I guess it could be somehow possible to incorporate the former into the latter, but my quick attempt in doing so resulted in a lot of compiler complains, so I think it would be a much bigger refactoring than we want to do right now.

>Why do we need, here, a []MachineNetwork{} and a []*models.MachineNetwork{}?

We use `[]MachineNetwork{}` in order to locally generate a part of install-config.yaml because this is a structure that stores `networking.MachineNetwork` part from this yaml. On the other hand `[]*models.MachineNetwork{}` is how we store it globally, as a `cluster` parameter. In general the piece of code we are touching here is a moment when `cluster.MachineNetwork` gets converted to the `networking.MachineNetwork` in install-config.yaml; as those two are represented by slightly different structures (because of the mentioned above `clusterID`), we have a mix of structs.

Just to be clear, I would happily collapse both of those into a single `something.MachineNetwork` type, but I don't see how to easily do it. Maybe @YuviGold has some ideas, though

Reply:
I looked as well if it is possible to merge these structs but I couldn't find how.
First, there's the `clusterID` field and possible other metadata fields we store in our DB.
Second, there's the `yaml` tags that aren't generated by swagger. (swagger automatically generates `json` tags).